editor/editor_file_dialog.cpp

```diff
@@ -1329,49 +1342,54 @@ void EditorFileDialog::_update_favorites() {
 	favorite->set_pressed(false);
 
 	Vector<String> favorited = EditorSettings::get_singleton()->get_favorites();
+	Vector<String> favorited_names;
+	favorited_names.resize(favorited.size());
 
 	bool fav_changed = false;
-	for (int i = favorited.size() - 1; i >= 0; i--) {
+	int current_favorite = -1;
+	for (int i = 0; i < favorited.size(); i++) {
 		if (!dir_access->dir_exists(favorited[i])) {
-			favorited.remove_at(i);
+			favorited.remove_at(i--);
 			fav_changed = true;
+		} else {
+			bool cres = favorited[i].begins_with("res://");
+			if (cres != res) {
+				continue;
+			}
+
+			// Compute favorite display text.
+			String name = favorited[i];
+			if (res && name == "res://") {
+				if (name == current) {
+					current_favorite = i;
+				}
+				name = "/";
+				favorited_names.write[i] = name;
+			} else if (name.ends_with("/")) {
+				if (name == current || name == current + "/") {
+					current_favorite = i;
+				}
+				name = name.substr(0, name.length() - 1);
+				name = name.get_file() + "/";
```

Comment:
@reduz suggests not showing the trailing slash, there's already a folder icon so it doesn't add much value.